### ISSUE TITLE
CMake: Doxygen not building

### ISSUE THREAD

**Issue by anonimal** [monero-project/kovri#868]
---
**By submitting this issue, I confirm the following:**

- I have read and understood the contributor guide in [kovri-docs](https://github.com/monero-project/kovri-docs).
- I have checked that the issue I am reporting can be replicated or that the feature I am suggesting is not present.
- I have checked opened or recently closed [pull requests](https://github.com/monero-project/kovri/pulls) for existing solutions/implementations to my issue/suggestion.
---

Introduced in #842, as seen [here](https://build.getmonero.org/builders/kovri-static-ubuntu-amd64/builds/475) and in other nightly builds:

```
-- Found Doxygen: /usr/bin/doxygen (found version "1.8.11") 
-- Found DOXYFILE_IN: /home/vagrant/slave/kovri-static-ubuntu-amd64/build/contrib/utils/doxygen/Doxyfile.in  
CMake Error at cmake/UseDoxygen.cmake:161 (get_target_property):
  get_target_property() called with non-existent target "doc".
Call Stack (most recent call first):
  CMakeLists.txt:316 (include)
```

Doxygen is only built for our nightly builds. I'd like to tweak our CI so we can catch things like this during PRs.

**Comment by anonimal** [comment]
ping @btolfa

**Comment by anonimal** [comment]
Resolved by #870.
